src/lib.rs

```diff
@@ -156,42 +158,47 @@ fn derive_property_for_field(field: &FieldDef) -> Vec<proc_macro2::TokenStream>
     }
     if let Some(ts) = field_conf.set.vis.to_ts().and_then(|visibility| {
         let method_name = field_conf.set.name.complete(field_name);
+        let (mut_ret_ty, ret_ty, ret_val) = if !field_conf.set.noret {
+            (quote!(&mut Self), quote!(Self), quote!(self))
+        } else {
+            (quote! {()}, quote! {()}, quote! {()})
+        };
         let generated = match prop_field_type {
             FieldType::Vector(inner_type) => match field_conf.set.typ {
                 SetTypeConf::Ref => quote!(
                     #visibility fn #method_name<T: Into<#inner_type>>(
                        &mut self,
                        val: impl IntoIterator<Item = T>
-                    ) -> &mut Self {
+                    ) -> #mut_ret_ty {
                         self.#field_name = val.into_iter().map(Into::into).collect();
-                        self
+                        #ret_val
                     }
                 ),
                 SetTypeConf::Own => quote!(
                     #visibility fn #method_name<T: Into<#inner_type>>(
                         mut self,
                         val: impl IntoIterator<Item = T>
-                    ) -> Self {
+                    ) -> #ret_ty {
                         self.#field_name = val.into_iter().map(Into::into).collect();
-                        self
+                        #ret_val
                     }
                 ),
             },
             _ => match field_conf.set.typ {
                 SetTypeConf::Ref => quote!(
                     #visibility fn #method_name<T: Into<#field_type>>(
                         &mut self, val: T
-                    ) -> &mut Self {
+                    ) -> #mut_ret_ty {
                         self.#field_name = val.into();
-                        self
+                        #ret_val
                     }
                 ),
                 SetTypeConf::Own => quote!(
                     #visibility fn #method_name<T: Into<#field_type>>(
                         mut self, val: T
-                    ) -> Self {
+                    ) -> #ret_ty {
```

Comment:
`#[property(set(noret, type="own"))]` is pointless.
Input is a owned self but no return.